command/unixOSCommand.go

```diff
@@ -46,8 +45,15 @@ func (unixOSCommand) CopyInlineFile(
 	useSudo bool,
 	opts ...pulumi.ResourceOption) (*remote.Command, error) {
 
-	createCmd := utils.WriteStringCommand(remotePath, useSudo)
-	return copyInlineFile(name, runner, fileContent, useSudo, createCmd, opts...)
+	sudo := ""
+	if useSudo {
+		sudo = "sudo"
+	}
+	backupPath := remotePath + "." + backupExtension
+	backupCmd := fmt.Sprintf("if [ -f '%v' ]; then %v mv -f '%v' '%v'; fi", remotePath, sudo, remotePath, backupPath)
+	createCmd := fmt.Sprintf("(%v) && cat - | %s tee %s > /dev/null", backupCmd, sudo, remotePath)
+	deleteCmd := fmt.Sprintf("if [ -f '%v' ]; then %v mv -f '%v' '%v'; else %v rm -f '%v'; fi", backupPath, sudo, backupPath, remotePath, sudo, remotePath)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Should you also remove the backup ? Would it collide with a new backup creation, or would it be overwritten ?

Reply:
because of `mv -f` old backup would be overwritten.